src/Http/Search/SearchQuery.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Werkspot\KvkApi\Http\Search;
+
+/**
+ * Class SearchQuery
+ *
+ * @author Patrick Development <info@patrickdevelopment.nl>
+ */
+final class SearchQuery implements QueryInterface
+{
+    /**
+     * KvK number, identifying number for a registration in the Netherlands Business Register. Consists of 8 digits
+     *
+     * @var string
+     */
+    private $kvkNumber;
+
+    /**
+     * Branch number (Vestigingsnummer), identifying number of a branch. Consists of 12 digits
+     *
+     * @var string
+     */
+    private $branchNumber;
+
+    /**
+     * RSIN is an identification number for legal entities and partnerships. Consist of only digits
+     *
+     * @var int
+     */
+    private $rsin;
+
+    /**
+     * Street of an address
+     *
+     * @var string
+     */
+    private $street;
+
+    /**
+     * House number of an address
+     *
+     * @var string
+     */
+    private $houseNumber;
+
+    /**
+     * Postal code or ZIP code, example 1000AA
+     *
+     * @var string
+     */
+    private $postalCode;
+
+    /**
+     * City or Town name
+     *
+     * @var string
+     */
+    private $city;
+
+    /**
+     * Indication  to include searching through inactive dossiers/deregistered companies.
+     * @note History of inactive companies is after 1 January 2012
+     *
+     * @var bool
+     */
+    private $includeInactiveRegistrations;
+
+    /**
+     * restrictToMainBranch Search is restricted to main branches.
+     *
+     * @var bool
+     */
+    private $restrictToMainBranch;
+
+    /**
+     * Defines the search collection for the query
+     *
+     * @var string
+     */
+    private $site;
+
+    /**
+     * User can optionally add a context to identify his result later on
+     *
+     * @var string
+     */
+    private $context;
+
+    /**
+     * Free format text search for in the compiled search description.
+     *
+     * @var string
+     */
+    private $q;
```

Comment:
I think we can improve the understandability of this variable and related methods by renaming it. We shouldn't need comments to understand what a variable or method is for.

In this case, `query` could be an alternative. Although it slightly clashes with the class name. I think it's a better alternative than `q`.

```suggestion
    private $query;
```

Reply:
solved